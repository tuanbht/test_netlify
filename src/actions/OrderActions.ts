import { CANCEL_ORDER_PATH, MARK_AS_PAID_PATH, ORDER_DETAILS_PATH } from '../constants/ApiPaths';
import { replace } from 'lodash';
import { Action } from 'redux';
import { CANCEL_ORDER, GET_ORDER_DETAILS, MARK_AS_PAID_ORDER } from '../constants/ReduxActions';
import { get, put } from './AxiosRequestActions';

const buildUrl = (url: string, orderId: number) => {
  return replace(url, ':id', String(orderId));
};

const getOrderDetailsAction = (orderId: number): Action => {
  return get(GET_ORDER_DETAILS, buildUrl(ORDER_DETAILS_PATH, orderId));
};

const cancelOrderAction = (orderId: number): Action => {
  return put(CANCEL_ORDER, buildUrl(CANCEL_ORDER_PATH, orderId));
};

const markAsPaidOrderAction = (orderId: number): Action => {
  return put(MARK_AS_PAID_ORDER, buildUrl(MARK_AS_PAID_PATH, orderId));
};

export default {
  getOrderDetailsAction,
  cancelOrderAction,
  markAsPaidOrderAction,
};
