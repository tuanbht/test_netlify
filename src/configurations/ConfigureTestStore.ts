import { configureStore, EnhancedStore } from '@reduxjs/toolkit';
import MockAdapter from 'axios-mock-adapter';
import axiosMiddleware from 'redux-axios-middleware';
import RootReducer from '../reducers';
import axiosClient from './ApiClient';

export const testStore = (initialState = {}): EnhancedStore =>
  configureStore({
    preloadedState: initialState,
    reducer: RootReducer,
    middleware: [axiosMiddleware(axiosClient)],
  });

export const mockAxios = new MockAdapter(axiosClient);
