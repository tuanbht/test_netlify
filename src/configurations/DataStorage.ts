import { ORDER_ID, PAYMENT_TOKEN } from '../constants/StoreKeys';

const setToken = (token: string): void => {
  localStorage.setItem(PAYMENT_TOKEN, token);
};

const setOrderId = (orderId: number): void => {
  localStorage.setItem(ORDER_ID, String(orderId));
};

const getToken = (): string | null => {
  return localStorage.getItem(PAYMENT_TOKEN);
};

const getOrderId = (): number => {
  return parseInt(localStorage.getItem(ORDER_ID) as string);
};

const clearAll = (): void => {
  localStorage.clear();
};

export default {
  setOrderId,
  setToken,
  getToken,
  getOrderId,
  clearAll,
};
