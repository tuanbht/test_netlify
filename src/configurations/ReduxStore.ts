import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import axiosMiddleware from 'redux-axios-middleware';
import axiosClient from './ApiClient';
import rootReducer from '../reducers';
import logger from 'redux-logger';

const history = createBrowserHistory();

const middleware = [
  axiosMiddleware(axiosClient),
  ...getDefaultMiddleware({
    serializableCheck: false,
  }),
  logger,
];

const store = configureStore({
  reducer: rootReducer,
  middleware: middleware,
});

export { history, store };
