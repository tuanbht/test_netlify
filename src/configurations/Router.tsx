import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { NOT_FOUND_PATH, ORDER_CANCELLED_PATH, ORDER_EXPIRED_PATH, ORDER_PATH, ROOT_PATH } from 'constants/RouterPaths';
import { ORDER_CANCELLED_CONTENT, ORDER_EXPIRED_CONTENT } from 'constants/NotificationContent';
import { CustomerPayments, NotFound, NotificationPage, HomePage } from 'containers';

export const Router = (): React.ReactElement => {
  return (
    <Switch>
      <Route exact path={ROOT_PATH} component={HomePage} />
      <Route exact path={ORDER_CANCELLED_PATH} render={() => <NotificationPage content={ORDER_CANCELLED_CONTENT} />} />
      <Route exact path={ORDER_EXPIRED_PATH} render={() => <NotificationPage content={ORDER_EXPIRED_CONTENT} />} />
      <Route exact path={ORDER_PATH} component={CustomerPayments} />
      <Route exact path={NOT_FOUND_PATH} component={NotFound} />
      <Route render={() => <Redirect to={NOT_FOUND_PATH} />} />
    </Switch>
  );
};

export default Router;
