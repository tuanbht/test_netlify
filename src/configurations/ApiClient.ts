import Axios from 'axios';
import { API_URL } from 'constants/ApiPaths';
import { NOT_FOUND_PATH } from 'constants/RouterPaths';
import { store } from './ReduxStore';

const axiosClient = Axios.create({
  baseURL: API_URL,
  headers: {
    Accept: 'application/vnd.api+json',
    'Content-Type': 'application/vnd.api+json',
  },
});

axiosClient.interceptors.request.use((config) => {
  const token = store.getState().Credential.token;
  if (token) config.headers['Payment-Token'] = token;
  else window.location.assign(NOT_FOUND_PATH);
  return config;
});

export default axiosClient;
