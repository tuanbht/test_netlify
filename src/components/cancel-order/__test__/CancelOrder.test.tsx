import { Button } from '@material-ui/core';
import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import CancelOrder from '../index';
import { MemoryRouter } from 'react-router-dom';
import { testStore } from '../../../configurations/ConfigureTestStore';
import { Provider } from 'react-redux';
import OrderActions from '../../../actions/OrderActions';
import faker from 'faker';

describe('CancelOrder', () => {
  const orderId = faker.random.number();
  const token = faker.random.uuid();

  const store = testStore({
    Credential: {
      token,
      orderId,
    },
  });
  let container: ReactWrapper;

  beforeEach(() => {
    jest.spyOn(store, 'dispatch');

    container = mount(
      <Provider store={store}>
        <MemoryRouter>
          <CancelOrder />
        </MemoryRouter>
      </Provider>,
    );
  });

  describe('click on quit crypto payment button', () => {
    beforeEach(() => {
      container.find(Button).find({ children: 'Quit Crypto Payment' }).first().simulate('click');
    });

    it('should call cancel order api', () => {
      expect(store.dispatch).toBeCalledWith(OrderActions.cancelOrderAction(orderId));
    });
  });
});
