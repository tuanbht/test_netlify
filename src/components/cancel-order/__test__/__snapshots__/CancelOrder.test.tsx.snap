// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelOrder renders template correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="makeStyles-content-2"
  >
    <div
      className="makeStyles-title-3"
    >
      Quit the payment system?
    </div>
    <div
      className="makeStyles-subtitle-4"
    >
      If you quit now, your crypto payment will be cancelled and youâ€™ll have to contact your merchant to complete the order with cash.
    </div>
    <Link
      className="makeStyles-cancelButton-5"
      to="/order-cancelled"
    >
      <WithStyles(ForwardRef(Button))
        classes={
          Object {
            "focusVisible": "makeStyles-focusVisible-7 makeStyles-focusVisible-9",
            "root": "makeStyles-root-6 makeStyles-root-8",
          }
        }
        onClick={[Function]}
      >
        Quit Crypto Payment
      </WithStyles(ForwardRef(Button))>
    </Link>
  </div>
</div>
`;
